api/v3/openapi.yaml

```diff
@@ -1179,6 +1221,95 @@ components:
         - DEAD
         - EMPTY
 
+    ConsumerLagData:
+      allOf:
+        - $ref: '#/components/schemas/Resource'
+        - type: object
+          required:
+            - cluster_id
+            - consumer_group_id
+            - current_offset
+            - log_end_offset
+            - lag
+            - consumer_id
+            - client_id
+            - topic_name
+            - partition_id
+            - partition
+          properties:
+            cluster_id:
+              type: string
+            consumer_group_id:
+              type: string
+            current_offset:
+              type: integer
+            log_end_offset:
+              type: integer
+            lag:
+              type: integer
+            consumer_id:
+              type: string
+            instance_id:
+              type: string
+              nullable: true
+            client_id:
+              type: string
+            topic_name:
+              type: string
+            partition_id:
+              type: integer
+            partition:
+              $ref: '#/components/schemas/Relationship'
```

Comment:
Oops, thanks for bringing this up. My original intention for the GetConsumerLagResponse call was to allow users to view lag of any topic-partition in a consumer group, so I'll have to rethink how I want to do this.

Reply:
This spec is looking good. And good catch @rigelbm about a consumer being subscribed to multiple partitions. In most cases, a consumer is subscribed to multiple partitions. 

Not with that said, this change does leave out some information. It is not possible to get the lag on an individual partition, or perhaps I'm not understanding how `PartitionData` works in the spec. Perhaps in `ConsumerLagDataList`, for a given consumer id, we can return an array containing partition id and lag?

Reply:
@addisonhuddy As you mentioned, I want ConsumerLagData to have finer granularity than just giving the information for the partition w/ the greatest lag. I like your idea of returning an array of partition ids and lag for the given consumer id, but am wondering if this is the most useful way to return lag for individual partitions. For instance, if a user is interested in finding the lag associated with a specific topic rather than a specific consumer, they would have to call GetConsumerLag on multiple consumer ids. (For this use case, it might be better to structure the GetConsumerLag query to accept `topic_name` and/or `partition_id` instead of `consumer_id`.)
Then again, I'm not sure if users would ever be concerned with a specific topic's lag over a specific consumer's lag, or do anything less than query lag for all partitions in a consumer group, in which case it wouldn't matter if they query based on consumer id or topic.

Reply:
The good news is this is something we can iterate on. We don't have to get it right the first go. As one consumer said to me a few weeks ago, "whatever lag metrics you give me, I'm just going to put them into New Relic."

From how I know customer's use per partition lag is by first passing in a consumer id.  They first see a consumer falling behind and then they might want to look up what is happening on a given partition.